solr/solrj/src/java/org/apache/solr/client/solrj/util/Constants.java

```diff
@@ -25,4 +25,15 @@ public class Constants {
   public static final boolean JRE_IS_MINIMUM_JAVA11 = true;
   // Future, enable if needed...
   // public static final boolean JRE_IS_MINIMUM_JAVA17 = Runtime.version().feature() >= 17;
+
+  public static final boolean IS_IBM_JAVA = isIBMJava();
+
+  private static boolean isIBMJava() {
+    try {
+      Class.forName("com.ibm.security.auth.module.Krb5LoginModule", false, null);
+      return true;
+    } catch (ClassNotFoundException e) {
```

Comment:
this works, but you really should also catch SecurityException, as this could happen with SecurityManager as enabled in SOlr by default.

Also to prevent a security Exception the last parameter should be `Constants.class.getClassLoader()` (this is what the default forName is doing). Sending null will use system classloader and that one is restricted by security manager.

Reply:
See also Javadoc: 

> If the parameter loader is null, the class is loaded through the bootstrap class loader.
> 
> [SecurityException](https://docs.oracle.com/en/java/javase/12/docs/api/java.base/java/lang/SecurityException.html) - if a security manager is present, and the loader is null, and the caller's class loader is not null, and the caller does not have the [RuntimePermission](https://docs.oracle.com/en/java/javase/12/docs/api/java.base/java/lang/RuntimePermission.html)("getClassLoader")

So to not get into security problems use the suggestion here:

> For example, in an instance method the expression:
> Class.forName("Foo")
> is equivalent to:
> Class.forName("Foo", true, this.getClass().getClassLoader())

so just replace by "false".

Reply:
yea `this.getClass().getClassLoader()` won't work with the static variable. `Constants.class.getClassLoader()` does.

Reply:
Fixed in e258862

Reply:
Thanks looks fine to me. You code including logging looks like I produced yesterday in Lucene two times - especially in static initializers you must be very careful with SecurityExceptions, they are unchecked and can appear at many places. The code then make class completely unuseable, which is horrible for some central, often used class like Constants - or the underlying directory implementation of Lucene: https://github.com/apache/lucene/blob/7f483bd6180c5c88176cc7c212405680d67ae249/lucene/core/src/java/org/apache/lucene/store/MMapDirectory.java#L376-L390